packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -167,3 +183,80 @@ class RenderSliverFillRemaining extends RenderSliverSingleBoxAdapter {
+/// there is never any room for anything beyond this sliver.
+///
+/// See also:
+///
+///  * [RenderSliverFillRemainingWithScrollable], which lays out its scrollable
+///    child without overscroll.
+///  * [RenderSliverFillRemainingWithoutScrollable], which lays out its
+///    non-scrollable child without overscroll.
+///  * [RenderSliverFillViewport], which sizes its children based on the
+///    size of the viewport, regardless of what else is in the scroll view.
+///  * [RenderSliverList], which shows a list of variable-sized children in a
+///    viewport.
+class RenderSliverFillRemainingAndOverscroll extends RenderSliverSingleBoxAdapter {
+  /// Creates a [RenderSliver] that wraps a non-scrollable [RenderBox] which is
+  /// sized to fit the remaining space plus any overscroll in the viewport.
+  RenderSliverFillRemainingAndOverscroll({
+    RenderBox child,
+  }) : super(child: child);
+
+  @override
+  void performLayout() {
+    double childExtent;
+    // The remaining space in the viewportMainAxisExtent. Can be <= 0 if we have
+    // scrolled beyond the extent of the screen.
+    double extent = constraints.viewportMainAxisExtent - constraints.precedingScrollExtent;
+    // The maxExtent includes any overscrolled area. Can be < 0 if we have
+    // overscroll in the opposite direction, away from the end of the list.
+    double maxExtent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
+
+    if (child != null) {
+      switch (constraints.axis) {
+        case Axis.horizontal:
+          childExtent = child.getMaxIntrinsicWidth(constraints.crossAxisExtent);
+          break;
+        case Axis.vertical:
+          childExtent = child.getMaxIntrinsicHeight(constraints.crossAxisExtent);
+          break;
+      }
+
+      // If the childExtent is greater than the computed extent, we want to use
+      // that instead of potentially cutting off the child. This allows us to
+      // safely specify a maxExtent.
+      extent = math.max(extent, childExtent);
+      // The extent could be larger than the maxExtent due to inverse
+      // overscrolling or a larger child size.
+      maxExtent = math.max(extent, maxExtent);
+      child.layout(constraints.asBoxConstraints(minExtent: extent, maxExtent: maxExtent));
+    }
+
+    assert(extent.isFinite,
+    'The calculated extent for the child of SliverFillRemaining is not finite.'
+      'This can happen if the child is a scrollable, in which case, the'
+      'hasScrollBody property of SliverFillRemaining should not be set to'
+      'false.',
+    );
+    final double paintedChildSize = calculatePaintOffset(constraints, from: 0.0, to: extent);
+    assert(paintedChildSize.isFinite);
+    assert(paintedChildSize >= 0.0);
+    geometry = SliverGeometry(
+      scrollExtent: extent,
+      paintExtent: math.min(maxExtent, constraints.remainingPaintExtent),
+      maxPaintExtent: math.min(maxExtent, constraints.remainingPaintExtent),
```

Comment:
Why's this maxPaintExtent bound by the remainingPaintExtent?

Reply:
Fixed.